eng/Generate.ps1

```diff
@@ -194,7 +194,11 @@ if (!($Exclude -contains "TestProjects"))
             continue
         }
         if ($testName.EndsWith("Cadl")) {
-            Add-Cadl $testName $directory "" "--option @azure-tools/cadl-csharp.generate-convenience-methods=false"
+            if ($testName -eq "GenerateAllModels-Cadl") {
```

Comment:
I think we need to add a cadl-project.yaml file that can take any options.  Then we can generically read and append any number of options instead of specific implementations for every option.

The issue with letting cadl compile directly consume this is we are using a common cadl installation at the root of the repo and the cadl-project.yaml will be inside each test directory.  I opened [this issue](https://github.com/microsoft/cadl/issues/1633) as a future enhancement at which point we can remove our parsing of the cadl-project.yaml file but until then this is probably the best path forward.

Reply:
The cadl compiler does not load cadl-project.yaml. There is some problem in our test framework and we are working it. This is a workaround before we resolve the test framework issue.

Reply:
> The cadl compiler does not load cadl-project.yaml

This is fixed. Would you try again?

Reply:
I add cadl-project.yaml to configure the flag.